packages/components/src/systems/SDKProvider/index.tsx

```diff
@@ -1,17 +1,36 @@
 import { PhotonClient } from '@photonhealth/sdk';
-import { createContext, JSXElement, useContext } from 'solid-js';
+import { createContext, createEffect, JSXElement, Ref, useContext } from 'solid-js';
+import { createStore } from 'solid-js/store';
 
-const SDKContext = createContext<PhotonClient>();
+const SDKContext = createContext<{ client: PhotonClient; ref: Ref<any> | undefined }>();
 
 interface SDKProviderProps {
   client: PhotonClient;
   children: JSXElement;
 }
 
 export default function SDKProvider(props: SDKProviderProps) {
-  return <SDKContext.Provider value={props.client}>{props.children}</SDKContext.Provider>;
+  let ref: Ref<any> | undefined;
+  const [store, setStore] = createStore({
+    client: props.client,
+    ref
+  });
+
+  createEffect(() => {
+    setStore({ client: props.client, ref });
+  });
+
+  return (
+    <SDKContext.Provider value={store}>
+      <div ref={ref}>{props.children}</div>
```

Comment:
Oh I DEFINITELY wouldn't do this. Do
```suggestion
      <div ref={ref} hidden />{props.children}
```

Reply:
Otherwise youre changing the tree and it can get real messy